clouddrift/analysis.py

```diff
@@ -4,6 +4,62 @@
 from clouddrift.haversine import distance, bearing
 
 
+def segment(
+    x: np.ndarray, tolerance: float, rowsize: np.ndarray[int] = None
+) -> np.ndarray[int]:
+    """Segment an array into contiguous segments.
+
+    Parameters
+    ----------
+    x : list, np.ndarray, or xr.DataArray
+        An array to segment.
+    tolerance : float
+        The maximum distance between consecutive points in a segment.
+    rowsize : np.ndarray[int], optional
+        The size of rows if x is a ragged array. If present, the function result
+        is a list of arrays. Otherwise, the function result is a single array.
+
```

Comment:
I am wondering if the function should not return instead a single array when the rowsize argument is present, just like when it is not. What is the point of getting a new rowsize? To be able to quickly form the new index `index=np.insert(np.cumsum(rowsize),0,0)` in order to apply function per segment as an example. Returning a list of array requires the user to concatenate the return list of array into a single array.

Reply:
Sounds good. It wasn't clear to me whether or not it's important for users to differentiate between gaps in data and gaps between trajectories. If it's not important, then our function result can always be of the same type, which is good design. This will be an easy change to make.